### PR TITLE
Add integration tests using `sls package`

### LINKED ISSUE #5938: Packaging - Create sophisticated integration test suite
Due to recent problems we had with https://github.com/serverless/serverless/pull/5297 problems (reverted via https://github.com/serverless/serverless/pull/5936) we should work on an integration test which thoroughly tests different packaging configuration (some of them are shared in the issues) to ensure that we won't break anything packaging-related in the future.

/cc @billfine @ganeshvlrk @eahefnawy @dschep

### PR DESCRIPTION
**Pull request by dschep**
## What did you implement:

Closes #5938 by adding integration tests that run `sls package`. 

It currently only covers very basic test cases. What cases should this test?

I currently use `sls create` but want to switch to configs provided by the test for easier to read & more reliable tdsts

## How did you implement it:
Ported the idea from https://github.com/UnitedIncome/serverless-python-requirements/blob/master/test.js but switched from tape to jest

## How can we verify it:
clone the branch and run `npm run local-integration-test`

## Todos:

- [x] Write tests
- [ ] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** NO
***Is it a breaking change?:*** NO

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "lint": "eslint . --cache",
     "docs": "node scripts/generate-readme.js",
     "integration-test-cleanup": "node scripts/integration-test-cleanup.js",
+    "local-integration-test": "jest --maxWorkers=5 local-suite",
```
Comment (medikoo) [inline package.json]: A suggeston: I would probably name it `package-integration-test` and those that deploy projects to AWS `deploy-integration-test`

File: tests/local-suite/packaging.tests.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
+const fs = require('fs');
```
Comment (medikoo) [inline tests/local-suite/packaging.tests.js]: `'use strict'` (I think it'll be good to turn on ESLint rule for that)

File: lib/plugins/create/templates/aws-alexa-typescript/serverless.yml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 service:
   name: aws-alexa-typescript
+#app: your-app-name
+#tenant: your-tenant-name
```
Comment (medikoo) [inline lib/plugins/create/templates/aws-alexa-typescript/serverless.yml]: Was that change intentional?

File: .travis.yml [hunk 4]
```diff
@@ -44,7 +40,8 @@ sudo: false
 install:
 - travis_retry npm install
 script:
-- if [[ -z "$INTEGRATION_TEST" && -z "$DISABLE_TESTS" ]]; then npm test; fi
+- npm i -g npm
```
Comment (medikoo) [inline .travis.yml]: Why is that needed?

File: tests/local-suite/packaging.tests.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+const path = require('path');
+const execSync = require('child_process').execSync;
+const fse = require('fs-extra');
+const testUtils = require('../utils/index');
+
+const serverlessExec = path.join(__dirname, '..', '..', 'bin', 'serverless');
+
+describe('simple packaging', () => {
```
Comment (pmuens) [inline tests/local-suite/packaging.tests.js]: Can we warp this into a global describe block like `Integration test - Packaging`?

File: tests/local-suite/packaging.tests.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        const nodeModules = new Set(
+          zipfiles.filter(f => f.startsWith('node_modules')).map(f => f.split(path.sep)[1]));
+        const nonNodeModulesFiles = zipfiles.filter(f => !f.startsWith('node_modules'));
+        expect(nodeModules).toEqual(new Set(['lodash']));
+        expect(nonNodeModulesFiles).toEqual(['handler.js']);
+      });
+  });
+});
+
+describe('cloudformation packaging', () => {
```
Comment (pmuens) [inline tests/local-suite/packaging.tests.js]: Can this also be wrapped in a "global" `Integration Test - Packaging` `describe` block?

File: .travis.yml (hunk 4 above)
Comment (pmuens) [reply .travis.yml]: I guess it was added becasue we're using `npm` in the `packaging` test in child processes. Not sure if a global install is necessay though.

File: lib/plugins/create/templates/aws-alexa-typescript/serverless.yml (hunk 3 above)
Comment (pmuens) [reply lib/plugins/create/templates/aws-alexa-typescript/serverless.yml]: And if so, are all templates updated? What about Azure and Google?

Comment (dschep) [reply lib/plugins/create/templates/aws-alexa-typescript/serverless.yml]: not in this pr. oops.

File: tests/packaging-suite/packaging.tests.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        const nodeModules = new Set(
+          zipfiles.filter(f => f.startsWith('node_modules')).map(f => f.split(path.sep)[1]));
+        const nonNodeModulesFiles = zipfiles.filter(f => !f.startsWith('node_modules'));
+        expect(nodeModules).toEqual(new Set(['lodash']));
+        expect(nonNodeModulesFiles).toEqual(['handler.js']);
+      });
+  });
+});
+
+describe('Integration test - Packaging', () => {
```
Comment (pmuens) [inline tests/packaging-suite/packaging.tests.js]: I think we can nest this and rename this to `CloudFormation files` or something like that. So the structure could look like this:

```
describe('Integration Test - Packaging', () => {
  // the regular / core tests
  describe('CloudFormation files', () => {
    // CloudFormation specific tests
  })
})
```

This way we have an easy to read and reason about log output.

Comment (dschep) [reply tests/packaging-suite/packaging.tests.js]: I actually removed the separation since i have tests that check both.

File: .travis.yml (hunk 4 above)
Comment (dschep) [reply .travis.yml]: removed
